### PR TITLE
feat(gcf-utils): some bots have large payloads not appropriate for cloud task payload 

### PR DESCRIPTION
**Pull request by bcoe**
We should add a test for this, but I thought it would be good to get a hot fix in for build cop quickly.

This also unblocks @azizsonawalla's work  as well, which requires that bots have `@types/sonic-boom` available.

Refs: https://github.com/googleapis/repo-automation-bots/issues/699 (tracking issue for test).

### CODE REVIEW

File: packages/gcf-utils/package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "@types/express": "^4.17.0",
     "@types/ioredis": "^4.14.8",
     "@types/lru-cache": "^5.1.0",
+    "@types/sonic-boom": "^0.7.0",
```
Comment (bcoe) [inline packages/gcf-utils/package.json:30]: gcf-utils had problems on the main branch do to exporting `@types/sonic-boom`, but only having it as dev dep.

File: packages/gcf-utils/src/gcf-utils.ts [hunk 2]
```diff
@@ -283,8 +287,13 @@ export class GCFBootstrapper {
   }
 
   gcf(
-    appFn: ApplicationFunction
+    appFn: ApplicationFunction,
+    wrapOptions?: WrapOptions
   ): (request: express.Request, response: express.Response) => Promise<void> {
+    let background = true;
+    if (wrapOptions) {
```
Comment (JustinBeckwith) [inline packages/gcf-utils/src/gcf-utils.ts:294]: 🐛 - If I pass WrapOptions here as `{}`, or maybe with a second property down the road, it will set the default of wrapOptions to false, while it looks like you want the default to be true.

Instead I'd suggest:
- changing line 291 to be `wrapOptions: WrapOptions = {}` 
- use extend to do:

```
const defaults = { wrapOptions: true };
const options = extend(true, defaults, wrapOptions, {});
```

Comment (bcoe) [reply packages/gcf-utils/src/gcf-utils.ts:294]: The interface requires that `WrapOptions`, if provided, has a boolean `background` property,  I don't understand how you're passing `{}`, and if you did the side effect would just be that `background` is falsy.

Comment (bcoe) [reply packages/gcf-utils/src/gcf-utils.ts:294]: so, the `if (wrapOptions) {` is just checking whether or not `wrapOptions` is `undefined`.

Comment (bcoe) [reply packages/gcf-utils/src/gcf-utils.ts]: updated the code to be less confusing.

File: packages/gcf-utils/test/gcf-bootstrapper.ts [hunk 3]
```diff
@@ -59,7 +64,7 @@ describe('GCFBootstrapper', () => {
 
     let enqueueTask: sinon.SinonStub;
 
-    beforeEach(async () => {
+    async function mockBootstrapper(wrapOpts?: WrapOptions) {
```
Comment (tbpg) [inline packages/gcf-utils/test/gcf-bootstrapper.ts:67]: Why the change from `beforeEach`?
